bullet/src/JointFeatures.cc

```diff
@@ -0,0 +1,456 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include "JointFeatures.hh"
+
+#include <algorithm>
+#include <sdf/Joint.hh>
+
+namespace ignition {
+namespace physics {
+namespace bullet {
+
+/////////////////////////////////////////////////
+double JointFeatures::GetJointPosition(
+    const Identity &_id, const std::size_t _dof) const
+{
+  (void) _dof;
+  double result = ignition::math::NAN_D;
+  if (this->joints.find(_id.id) != this->joints.end())
+  {
+    const JointInfoPtr &jointInfo = this->joints.at(_id.id);
+    const int jointType = jointInfo->constraintType;
+    // Check the type of joint and act accordignly
+    if (jointInfo->constraintType ==
```

Comment:
Is the intent to handle other constraints?  In that case I think a switch makes more sense here.

Reply:
Yes, I changed it to a switch statement with 7d6d9d2